waypointy/oc/xml2sql.php

```diff
@@ -1,166 +1,350 @@
+                print("error sql : id:$id - query:$sql - mysqli_error:" . mysqli_error($link) . "\n");
+            }
+            $nDeleted++;
+            continue;
+        }
 
-$BAZY_OC['nl']['prefix'] = 'OB';
-$BAZY_OC['nl']['url'] = "http://www.opencaching.nl/xml/ocxml11.php?cache=1&session=0&charset=utf-8&cdata=1&xmldecl=1&ocxmltag=1&doctype=0&zip=0&modifiedsince=";
-$BAZY_OC['nl']['szukaj'] = 'http://www.opencaching.nl/searchplugin.php?userinput=';
+        // Check for needed fields and make an update
+        $sqlInsert = array();
+        $sqlValues = array();
+        $sqlUpdate = array();
 
+        if (isset($change->data->names)) {
+            $name = mysqli_real_escape_string($link, implode(' | ', (array)$change->data->names));
+            $sqlInsert [] = 'name';
+            $sqlValues [] = $name;
+            $sqlUpdate [] = "name=?";
+        }
+        if (isset($change->data->owner->username)) {
+            $owner = mysqli_real_escape_string($link, (string)$change->data->owner->username);
+            $sqlInsert [] = 'owner';
+            $sqlValues [] = $owner;
+            $sqlUpdate [] = "owner=?";
+        }
+        if (isset($change->data->location)) {
+            $location = explode('|', $change->data->location);
+            $lon = floatval($location[0]);
+            $lat = floatval($location[1]);
+
+            $sqlInsert [] = 'lon';
+            $sqlValues [] = $lon;
+            $sqlUpdate [] = "lon=?";
+
+            $sqlInsert [] = 'lat';
+            $sqlValues [] = $lat;
+            $sqlUpdate [] = "lat=?";
+        }
+        if (isset($change->data->type)) {
+            $type = mysqli_real_escape_string($link, (string)$change->data->type);
+            $sqlInsert [] = 'typ';
+            $sqlValues [] = $type;
+            $sqlUpdate [] = "typ=?";
+        }
+        if (isset($change->data->country)) {
+            $country = mysqli_real_escape_string($link, (string)$change->data->country);
+            $sqlInsert [] = 'kraj';
+            $sqlValues [] = $country;
+            $sqlUpdate [] = "kraj=?";
+        }
+        if (isset($change->data->url)) {
+            $url = mysqli_real_escape_string($link, (string)$change->data->url);
+            $sqlInsert [] = 'link';
+            $sqlValues [] = $url;
+            $sqlUpdate [] = "link=?";
+        }
 
-/*
-// tylko odkomentować //
-$BAZY_OC['cz']['prefix'] = 'OZ';
-$BAZY_OC['cz']['url'] = "/home/geokrety/public_html/tools/oc-cz.xml";
-$BAZY_OC['cz']['szukaj'] = 'http://www.opencaching.cz/searchplugin.php?userinput=';
-*/
+        if (sizeof($sqlInsert) > 0) {
+            // It can happen that changelog contains useless fields like "founds" which we are not interested in.
+            // So we need to trigger actual update only if at least one of our fields was changes.
 
+            $sqlInsert [] = 'waypoint';
+            $sqlValues [] = $id;
+            $sqlUpdate [] = "waypoint=?";
 
-include_once("../../konfig-tools.php");
```

Comment:
This include is necessary, as it define `$geokrety_www`

Reply:
oops